### ISSUE TITLE
ALB attributes are updated with previous values on attribute-only change

### ISSUE THREAD

**Issue by gws** [kubernetes-sigs/aws-load-balancer-controller#1185]
Hello,

When updating the `alb.ingress.kubernetes.io/load-balancer-attributes` annotation on an Ingress resource, the load balancer attributes that get set on the ALB appear to be the **previously-applied** set of attributes. To illustrate what I mean, I'll show log output from the ingress controller along with the actions I'm taking on the Ingress resource. The attributes start out with `idle_timeout.timeout_seconds=360` after the controller boots, which is the start of the log shown:

```
Mar 08 10:28:11 I0308 17:27:43.838943       1 controller.go:134] kubebuilder/controller "level"=0 "msg"="Starting Controller"  "controller"="alb-ingress-controller"
Mar 08 10:28:11 I0308 17:27:43.939164       1 controller.go:154] kubebuilder/controller "level"=0 "msg"="Starting workers"  "controller"="alb-ingress-controller" "worker count"=1
Mar 08 10:28:11 I0308 17:27:48.030695       1 controller.go:236] kubebuilder/controller "level"=1 "msg"="Successfully Reconciled"  "controller"="alb-ingress-controller" "request"={"Namespace":"my-ns","Name":"my-ingress"}
## I modify the attributes to have idle_timeout.timeout_seconds=600. Nothing happens.
Mar 08 10:28:11 I0308 17:27:52.661266       1 store.go:223] updating annotations information for ingress my-ns/my-ingress
Mar 08 10:28:11 I0308 17:27:53.161467       1 controller.go:236] kubebuilder/controller "level"=1 "msg"="Successfully Reconciled"  "controller"="alb-ingress-controller" "request"={"Namespace":"my-ns","Name":"my-ingress"}
## I modify the attributes back to idle_timeout.timeout_seconds=360. Immediately, I see the following log entry. Note that this value was my previous, unapplied change.
Mar 08 10:28:11 I0308 17:28:04.243724       1 store.go:223] updating annotations information for ingress my-ns/my-ingress
Mar 08 10:28:11 I0308 17:28:04.428135       1 attributes.go:144] my-ns/my-ingress: Modifying ELBV2 attributes to [{    Key: "idle_timeout.timeout_seconds",    Value: "600"  }].
Mar 08 10:28:11 I0308 17:28:04.818894       1 controller.go:236] kubebuilder/controller "level"=1 "msg"="Successfully Reconciled"  "controller"="alb-ingress-controller" "request"={"Namespace":"my-ns","Name":my-ingress"}
## I modify the attributes back to idle_timeout.timeout_seconds=600. Immediately, I see the following log entry. Note that this value was my previous, unapplied change.
Mar 08 10:28:25 I0308 17:28:24.744881       1 store.go:223] updating annotations information for ingress my-ns/my-ingress
Mar 08 10:28:25 I0308 17:28:24.904487       1 attributes.go:144] my-ns/my-ingress: Modifying ELBV2 attributes to [{    Key: "idle_timeout.timeout_seconds",    Value: "360"  }].
Mar 08 10:28:25 I0308 17:28:25.269518       1 controller.go:236] kubebuilder/controller "level"=1 "msg"="Successfully Reconciled"  "controller"="alb-ingress-controller" "request"={"Namespace":"my-ns","Name":"my-ingress"}
```

I'm not setting these attributes quickly or anything like that, I'm giving the controller ample time to respond before making changes. When the (incorrect) actions are taken by the Ingress controller, they happen promptly in response to my changes to the Ingress, it's just that they're the "previous" set of attributes getting set.

Further, when I add or modify a *different* annotation on the Ingress controller, it correctly picks up the current `load-balancer-attributes` on the Ingress resource and sets them in addition to any modifications required by the other annotation, so this just appears to be an issue when modifying the `load-balancer-attributes` by themselves.

I've noticed this on v1.1.3, v1.1.4, and v1.1.5. I haven't tested prior versions.

I believe this is similar to https://github.com/kubernetes-sigs/aws-alb-ingress-controller/issues/898 and it may be that restarting the controller pod would correctly set the ALB attributes. However, I think noting that the attributes are consistently updated to the "previous" set of attributes is interesting and wasn't mentioned in that report.

**Comment by gmauleon** [comment]
Saw a similar behavior on a 1.15 cluster. I was also setting the `timeout`, in case it's related.
Interestingly it also happens if you remove the `"alb.ingress.kubernetes.io/load-balancer-attributes"` annotation (it will keep the last value until another change is made to the ingress).

**Comment by gws** [comment]
/remove-lifecycle stale

I continue to see this behavior as recently as v1.1.8.

**Comment by M00nF1sh** [comment]
Hi guys,
These are caused by race condition in annotation handling in our version.

we have released a new v2.0.0 version which should have solved these issues, please try it out 😄 

Closing this, feel free to reopen if the issue exists in v2.0.0
